presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/ValidateDependenciesChecker.java

```diff
@@ -598,6 +599,25 @@ public Void visitApply(ApplyNode node, Set<Symbol> boundSymbols)
             return null;
         }
 
+        @Override
+        public Void visitLateralJoin(LateralJoinNode node, Set<Symbol> boundSymbols)
```

Comment:
shouldn't you have something similar to `NoApplyNodeLeftChecker`. No `LateralJoinNode` should exists in plan after all transformations.

Does this method actually get called since no `LaterlJoinNodes` should exists?